### ISSUE TITLE
how to use FSAM with grad scaler

### ISSUE THREAD

**Issue by stylebased** [nblt/F-SAM#5]
Hi,
When I use the mix precision with autocast in model training, how should I embed the fsam module with the grad_scaler with 'scale','unscale','clip_grad_norm' , 'step' and 'update' in grad_scaler?
Thanks

**Comment by nblt** [comment]
Hi,
While I recommend using bfloat16 to avoid the need for loss scaling, I suggest following these steps when embedding FSAM:

1. Scale the loss for the first gradient backpropagation of FSAM.
2. Unscale the gradients.
3. Perform the first gradient ascent step of FSAM.
4. Scale the loss for the second gradient backpropagation of FSAM.
5. Unscale the gradients.
6. Apply gradient clipping.
7. Perform the second gradient descent step of FSAM.
